beacon_node/beacon_chain/src/migrate.rs

```diff
@@ -709,12 +619,8 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> BackgroundMigrator<E, Ho
                 ]
             })
             .chain(abandoned_states.into_iter().flat_map(|(slot, state_hash)| {
-                [StoreOp::DeleteState {
-                    state_root: state_hash.into(),
-                    state_slot: Some(slot),
-                    // Abandoned forks will never be used, so we can safely delete the diffs
-                    prune_hot_diff: true,
-                }]
+                // Abandoned forks will never be used, so we can safely delete the diffs
```

Comment:
Noting this now includes:

- Abandoned forks (not descended from finalized)
- Old stuff that wasn't pruned previously but is getting pruned now (older than previous finalization)
- Finalized states from the canonical chain which are not required by the HDiff mechanism.

Reply:
Updated the comments